pkg/frontend/querymiddleware/astmapper/instant_splitting.go

```diff
@@ -62,6 +62,17 @@ var splittableRangeVectorAggregators = map[string]bool{
 	sumOverTime:   true,
 }
 
+// cannotDoubleCountBoundaries is the list of functions that cannot double count the boundaries points when being split by range.
+// Consider a timeline with samples at 1s,2s,3s,4s,5s...
+// Prometheus ranges are closed on both ends, so range [2s,4s] includes samples at 2s and 4s.
+// If we split that into [2s, 3s] and [3s, 4s], then we will include the sample at 3s in both ranges.
+// This cannot be done for some sample-aggregating functions like `count_over_time`: we would count 4 samples (2s,3s,3s,4s) after splitting.
+// However, for other functions, like `rate`, it is important to have both ends includes, as the rate is (start-end)/range.
+var cannotDoubleCountBoundaries = map[string]bool{
+	countOverTime: true,
```

Comment:
[question] Given we rewrite the `avg_over_time()` as `sum_over_time() / count_over_time()`, does this fix applies to that rewrite too? Is it covered by the the test changes you did?

Reply:
Yes, that will be rewritten with this fix. And it should be covered in the tests as we are also testing the `avg_over_time` expression with the different time dates 👍